src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -0,0 +1,136 @@
+                var intVal = Integer.parseInt(i.priority())
+                if (intVal == consecVals) consecVals = consecVals + 1
+            }
+        }
+        if (consecVals >= 1000) throw Exception("The upper limit of 1000 rules has been reached for this load balancer.")
+        return consecVals;
+    }
+
+    fun taskDefinitionWithOverride(ecs_cluster_name: String, emr_cluster_host_name: String, albName :String, user_name: String, jupyterCpu : Int=512, jupyterMemory: Int = 512) {
+
+        val credentials: AwsCredentialsProvider = credentialsService.getDefaultCredentialsProvider()
+
+        val ecsClient: EcsClient = EcsClient.builder().credentialsProvider(credentials).region(awsRegion).build()
+        val albClient: ElasticLoadBalancingV2Client = ElasticLoadBalancingV2Client.builder().credentialsProvider(credentials).region(awsRegion).build()
+
+        println("Getting alb information...")
+
+        val albRequest = DescribeLoadBalancersRequest.builder().names("orchestration-service-lb").build()
+        val albResponse: DescribeLoadBalancersResponse = albClient.describeLoadBalancers(albRequest)
+
+        println("Getting Listener information...")
+
+        val albListenerRequest = DescribeListenersRequest.builder().loadBalancerArn(albResponse.loadBalancers()[0].loadBalancerArn()).build()
+        val albListenerResponse = albClient.describeListeners(albListenerRequest)
+
+        println("Creating target group...")
+
+        val tgRequest = CreateTargetGroupRequest.builder().name("$user_name-target-group").protocol("HTTPS").vpcId(albResponse.loadBalancers()[0].vpcId()).port(443).build()
+        val targetGroupResponse = albClient.createTargetGroup(tgRequest)
+
+        println("Getting target group arn...")
+
+        val albTargetGroupRequest = albClient.describeTargetGroups(DescribeTargetGroupsRequest.builder().names("$user_name-target-group").build())
+        val albTargetGroupArn = albTargetGroupRequest.targetGroups()[0].targetGroupArn()
+
+        val pathPattern = PathPatternConditionConfig.builder().values("/$user_name/*").build()
+        val albRuleCondition :RuleCondition = RuleCondition.builder().field("path-pattern").pathPatternConfig(pathPattern).build()
+        val userTargetGroup = TargetGroupTuple.builder().targetGroupArn(albTargetGroupArn).build()
+        val forwardAction = ForwardActionConfig.builder().targetGroups(userTargetGroup).build()
+        val albRuleAction = Action.builder().type("forward").forwardConfig(forwardAction).build()
+
+        println("Creating listener rule...")
+
+        var rulesResponse = albClient.describeRules(DescribeRulesRequest.builder().listenerArn(albListenerResponse.listeners()[0].listenerArn()).build())
+        val createRule = albClient.createRule(CreateRuleRequest.builder().listenerArn(albListenerResponse.listeners()[0].listenerArn()).priority(getPriorityVal(rulesResponse)).conditions(albRuleCondition).actions(albRuleAction).build())
```

Comment:
This doesn't need to be loaded into a val.
Also same comment about taking the first in an array - are we sure it's the one we want?